File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def lcm(a, b):
    for i in range(max(a, b), (a * b) + 1):
        if i % a == 0 and i % b == 0:
            return i
        
def solution(answers):
    answer = []
    test = len(answers) # 정답의 개수 
    # 각 학생들의 반복 배열
    student1 = [1, 2, 3, 4, 5]
    student2 = [2, 1, 2, 3, 2, 4, 2, 5]
    student3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    students = [0, 0, 0]

    get_lcm = lcm(40, test)  # 최소 공배수를 구해줌 => 40

    student1_loop = student1 * 2000
    student1_loop = student1_loop[:test]
    student2_loop = student2 * 1250
    student2_loop = student2_loop[:test]
    student3_loop = student3 * 1000
    student3_loop = student3_loop[:test]

    for i in range(test):
        if (answers[i] == student1_loop[i]):
            students[0] += 1
        if (answers[i] == student2_loop[i]):
            students[1] += 1
        if (answers[i] == student3_loop[i]):
            students[2] += 1

    for i in range(3):
        if max(students) == students[i]:
            answer.append(i+1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    // 리턴 타입 꼭 맞춰주셔요 런타임 에러,,나용,,
    public ArrayList<Integer> solution(int[] answers) {
        // 쉣,, 자바 왜 배열 곱하기 안되녀,,ㅠㅠ
		ArrayList<Integer> answer = new ArrayList<>();
        
		int[] student1 = { 1, 2, 3, 4, 5 }; // 5개
		int[] student2 = { 2, 1, 2, 3, 2, 4, 2, 5 }; // 8개
		int[] student3 = { 3, 3, 1, 1, 2, 2, 4, 4, 5, 5 }; // 10개

		int[] students = { 0, 0, 0 };

		for (int i = 0; i < answers.length; i++) {
			if (answers[i] == student1[i % 5]) {
				students[0]++;
			}if (answers[i] == student2[i % 8]) {
				students[1]++;
			}if (answers[i] == student3[i % 10]) {
				students[2]++;
			}
		}
		// System.out.println(Arrays.toString(students));
		// 최고 개수
		int maxTest = Math.max(students[0], Math.max(students[1], students[2]));
		//System.out.println(maxTest);
		for (int i = 0; i < 3; i++) {
			if (students[i] == maxTest) {
				answer.add(i + 1);
			}
		}
		// 0빼기
		answer.remove(Integer.valueOf(0));

		return answer;
    }
}